### PR TITLE
[NO JIRA]: Migrate to gh actions

### PR DESCRIPTION
**Pull request by olliecurtis**
Migrating from Travis CI to GH Actions.

One last outstanding item is to create the tokens to allow docs to be deployed once built.

### CODE REVIEW

**Review by georgegillams — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    name: Trigger docs Travis CI build
+    if: github.ref == 'refs/heads/master'
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Trigger docs Travis CI
+        env:
+          TRAVIS_API_KEY: ${{ secrets.TRAVIS_API_KEY }}
+        run: |
+          curl -f -s -X POST -H "Content-Type: application/json" -H "Accept: application/json" -H "Travis-API-Version: 3" -H "Authorization: token $TRAVIS_API_KEY" -d '{"request": {"branch":"master"}}' https://api.travis-ci.org/repo/Skyscanner%2Fbackpack-docs/requests
```
[the remark this answers is not in the document]
Comment (olliecurtis) [reply .github/workflows/ci.yml:75]: Ok awesome will keep it in for now until that or is merged, just incase this goes in first so still need it 🙂

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+      run: npm ci
+    
+    - name: Build Backpack
+      run: npm run build
+
+    - name: Check package-lock
+      run: ./scripts/check-pristine-state package-lock.json
+    
+    - name: Test
+      run: (set -o pipefail; npm test |& tee test.log)
```
[the remark this answers is not in the document]
Comment (olliecurtis) [reply .github/workflows/ci.yml:34]: I can change it, this was just taken directly out of the travis step so left it so as not to case issues or loss of info 🙂

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+    
+    - name: nvm install
+      shell: bash -l {0}
```
Comment (shaundon) [inline .github/workflows/ci.yml:17]: What does this line do?

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -0,0 +1,75 @@
+name: Backpack CI
+
+on:
+  push:
```
Comment (k0nserv) [inline .github/workflows/ci.yml:4]: What about tags? Should we run on those too?(in which case the checks with `refs/head/master` will need to be tweaked as the ref will be `refs/tags/x.y.z`

Comment (olliecurtis) [reply .github/workflows/ci.yml:4]: Sure thing! Wasn't sure about this one as when we publish tags we publish multiple ones (one for each package) so not sure how that will play with the deploy based on tags as we don't want to kick off 30 builds at once 🤔

Comment (k0nserv) [reply .github/workflows/ci.yml:4]: That is a good point 🤔 Probably wouldn't be great if that happened

File: .github/workflows/ci.yml (hunk 3 above)
Comment (olliecurtis) [reply .github/workflows/ci.yml:17]: That is a good question I need to ask @k0nserv as I took this from backpack-ios GH Actions

Comment (shaundon) [reply .github/workflows/ci.yml:17]: Ah ok, yeah I'm sure it's there for a good reason, it just wasn't immediately obvious (in which case maybe we should pop a comment above it)

Comment (k0nserv) [reply .github/workflows/ci.yml:17]: See https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#using-a-specific-shell

File: .github/workflows/ci.yml (hunk 4 above)
Comment (olliecurtis) [reply .github/workflows/ci.yml:4]: Yeah we need to think how to do this responsibly - I took the root of just doing pushes based on master as we don't tend to make changes directly on master without them going through a PR so means if we are merging they should be good to deploy the docs straight away?

File: .github/workflows/ci.yml (hunk 3 above)
Comment (shaundon) [reply .github/workflows/ci.yml:17]: It says `bash` is the default though, so what does this line actually do that we don't already get built in?

Comment (olliecurtis) [reply .github/workflows/ci.yml:17]: We add the `-l` which from official bash docs says

```
-l        Make bash act as if it had been invoked as a login shell (see
                 INVOCATION below).
```

Comment (shaundon) [reply .github/workflows/ci.yml:17]: Why do we need to set this?

Comment (olliecurtis) [reply .github/workflows/ci.yml:17]: Referring again to @k0nserv as he set this up in iOS - guessing there was some issues with setting it up? I was planning to chat to him about some of these things to understand more what they meant

Comment (olliecurtis) [reply .github/workflows/ci.yml]: So tried removing it and it seems to fail, I think using the logged in parameter allows it to assume a state of a logged in shell so has access for tooling?

Screenshot 2020-10-27 at 11 23 36

Comment (shaundon) [reply .github/workflows/ci.yml]: Ah fair enough! Thanks for checking it

**Review by shaundon — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+    
+    - name: Deploy storybook
+      uses: peaceiris/actions-gh-pages@v3
+      if: github.ref == 'refs/heads/master'
+      with:
+        personal_token: ${{ secrets.DEPLOY_TOKEN }}
+        publish_dir: dist-storybook/
+        keep_files: true
+        external_repository: backpack/storybook
+        publish_branch: master
```
Comment (georgegillams) [inline .github/workflows/ci.yml:62]: Can we have a new line at the end of the file?

**Review by georgegillams — APPROVED** [review]

File: .github/workflows/ci.yml (hunk 5 above)
Comment (olliecurtis) [reply .github/workflows/ci.yml:62]: Merged before I saw this, will add something to master now :)

**Pull request merged**